include/highfive/bits/H5Node_traits.hpp

```diff
@@ -152,6 +152,17 @@ class NodeTraits {
     /// \param node_name The entry to check, path relative to the current group
     inline ObjectType getObjectType(const std::string& node_name) const;
 
+    void createSoftLink(const std::string& objPath,
```

Comment:
I don't see the definition/impl of hard links as mentioned by this PR

Reply:
@ferdonline agree but I thought that there is no need to add hard link as hard links are those that are created in usual way: for example `file.createGroup(...)` should create hard link to the group (if I'm not mistaken). How do you think, should I add method `createHardLink(...)`?

Reply:
Hard links are useful to add additional paths to datasets, or even move, see #357
Anyway its implementation can be a little bit more complicated and its urgency might be lower. So we will go with soft and external, while Hard links will be implemented in another PR.